Dockerfile

```diff
@@ -1,22 +1,38 @@
-FROM debian:latest
+ARG DEBIAN_VERSION=12
+
+FROM debian:${DEBIAN_VERSION}
+
+ENV DEBIAN_VERSION=12
+
+ENV DEBIAN_FRONTEND=noninteractive
+
 SHELL ["/bin/bash", "-c"]
-RUN source /etc/os-release && apt-get update && apt-get install -y curl wget \
-&& wget -q https://packages.microsoft.com/config/debian/$VERSION_ID/packages-microsoft-prod.deb \
+
+RUN apt-get update -y && apt-get install -y curl wget \
+&& wget https://packages.microsoft.com/config/debian/${DEBIAN_VERSION}/packages-microsoft-prod.deb \
 && dpkg -i packages-microsoft-prod.deb \
 && rm packages-microsoft-prod.deb \
-&& apt-get install -y locales curl wget nmap nano vim msmtp msmtp-mta python3-full screen irssi git testdisk tcpdump tshark \
-python3-pip zsh emacs python3-full python3-pip git elinks emacs rclone rsync mc zip unzip unar p7zip-full iperf3 mtr \
-&& pip3 install pipx --break-system-packages \
+&& apt-get install -y nix-bin locales curl wget nmap nano vim msmtp msmtp-mta python3-full screen irssi git testdisk tcpdump tshark \
+python3-pip zsh python3-full python3-pip pipx git elinks emacs rclone rsync mc zip unzip unar p7zip-full iperf3 mtr \
 && rm -rf /var/lib/apt/lists/*  && localedef -i en_US -c -f UTF-8 -A /usr/share/locale/locale.alias en_US.UTF-8 \
-&& mkdir -p /home/linuxbrew/ && chmod -Rv 777 /home/linuxbrew/ && useradd -ms /usr/bin/zsh blueteam
+&& mkdir -p /home/linuxbrew/ && chmod -Rv 777 /home/linuxbrew/ && useradd -ms /usr/bin/zsh blueteam \
+&& mkdir -p /nix && chown -R blueteam:blueteam /nix
+
+
 USER blueteam
+
 WORKDIR /home/blueteam
+
 # FROM HERE WE START WHAT THE USER SPACE LOOKS LIKE
+
+# Install nix channels and update
+RUN nix-channel --add https://nixos.org/channels/nixpkgs-unstable nixpkgs && nix-channel --update
```

Comment:
how much space does this take up? seems pretty small, didn't see it in the test run.